### PR TITLE
MINOR: Add Kafka Streams upgrade workflow

### PR DESCRIPTION
**Pull request by bbejeck**

### CODE REVIEW

**Comment by bbejeck** [comment]
Added upgrade workflow.

\cc @mjsax @guozhangwang @dguy

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -44,6 +44,20 @@
         See <a href="#streams_api_changes_0101">below</a> a complete list of 0.10.1 API changes that allow you to advance your application and/or simplify your code base, including the usage of new features.
     </p>
 
+    <h3><a id="full_upgrade_workflow" href="#full_upgrade_workflow">Full Upgrade Workflow</a></h3>
```
Comment (guozhangwang) [inline docs/streams/upgrade-guide.html:47]: Do we want to recommend this pattern for all the upgrade paths though, as it is not a pleasant path compared with rolling bounces (i.e. your app will be offline for a bit); I think for most of the upgrade path rolling bounce is still possible?

Comment (mjsax) [reply docs/streams/upgrade-guide.html:47]: I agree with Guozhang. We can leave this PR open for now and should look into our upgrade system test -- after we verify what work, we should update the docs accordingly. I think, this whole "stop the app completely" thing, is only required if you upgrade from `0.10.0` -- in general, rolling bounce upgrades should work.

Comment (bbejeck) [reply docs/streams/upgrade-guide.html:47]: Ack

**Comment by guozhangwang** [comment]
This is still an ongoing PR? @bbejeck @mjsax

**Comment by bbejeck** [comment]
I think so, but apparently, it fell through the cracks.  It looks to me we need to verify from system tests rolling upgrades work and which versions are eligible for rolling upgrades (from @mjsax's comment above looks to be 10.0.1+).  If don't have an existing systems test for this I know there is a Jira for testing rolling upgrades against a fixed broker version.

**Comment by guozhangwang** [comment]
@bbejeck Cool, let's come back to update the web docs in this PR after we have done the compatibility system tests.

**Comment by mjsax** [comment]
I think rolling bounce should work from `0.10.1.0` onwards. We have an incompatible change from `0.10.0.x` to `0.10.1.0` as we bumped the magic number in the rebalance user data but did not write compatible code...

**Comment by mjsax** [comment]
@bbejeck Just went through older PR. With KIP-268 this seems to become outdated. Should we close this PR? WDYT?

**Comment by bbejeck** [comment]
Was just about to ping you about the same thing, closing this now.

**Pull request closed without merging**